apps/examples/testcase/le_tc/kernel/tc_debug.c

```diff
@@ -0,0 +1,89 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file tc_debug.c
+
+/// @brief Test Case Example for Debug API
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <tinyara/debug/sysdbg.h>
+#include "tc_internal.h"
+
+#define BUFLEN 32
+
+static void tc_debug_sysdbg_write(void);
+
+#ifdef CONFIG_TC_KERNEL_DEBUG
+/**
+ * @fn                  :tc_debug_sysdbg_write
+ * @brief               :fops to read data buffer passed from user space
+ * @scenario            :fops to read data buffer passed from user space, return value should be equal to 0
+ * API's covered        :sysdbg_open, sysdbg_close, tc_debug_sysdbg_write
+ * Preconditions        :none
+ * Postconditions       :none
+ * @return              :void
+ */
+static void tc_debug_sysdbg_write(void)
+{
+	int ret = -1;
+	char buffer[BUFLEN];
+
+	int fd = open(SYSDBG_PATH, O_WRONLY);
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	strncpy(buffer, "enable_monitor", 14);
```

Comment:
Regardless of this PR, usages of sysdbg looks not good. Who can remember the command with string.
How about defining some values for all of commands in public sysdbg header?
And I can't understand what commands mean from the `show_usages` function in sysdbg.c.

Reply:
Normally, we use the length of the string with NULL termination.
But in this PR, and in sysdbg.c, we handles the strings without NULL termination.
Could you tell me why?

Reply:
@JeonginKim don't know why they used the strings without NULL termination, i am using it without NULL termination because it is working fine.

Reply:
Yeah, that's why I said `regardless of this PR`. In this PR, you can keep current usages. But with another PR, let's modify it.